wrangler-core/src/main/java/io/cdap/directives/parser/JsParser.java

```diff
@@ -143,8 +167,19 @@ public void destroy() {
               row.add(column, getValue(element.getAsJsonPrimitive()));
             }
           }
-        } catch (JSONException e) {
-          throw new ErrorRowException(NAME, e.getMessage(), 1);
+        } catch (Exception e) {
+          String msg = e.getCause() != null ? e.getCause().getMessage() : e.getMessage();
+          // In case there are more rows being handled, we attempt to surface the Json that is
+          // causing an issue to make it easier for users to identify the problem quickly.
+          if (rows.size() > 1) {
```

Comment:
isn't rows always size 1? Otherwise throwing an ErrorRowException would incorrectly mark every Row as an error?

Reply:
Not always. If we parsing AvroFile. Parse-as-avro-file will generate records > 1.

Reply:
yes, but the RecipeExecutor will break that into multiple lists of size 1. Either this is always size 1, or the entire ErrorRowException contract is wrong.